src/agent.jl

```diff
@@ -37,60 +37,45 @@ function iterate_agents_cb(agent::HSA.Agent, agents)
     return HSA.STATUS_SUCCESS
 end
 
-"Determines if a memory region can be used for kernarg allocations."
-function get_kernarg_memory_region_cb(region::HSA.Region, data::Ptr{HSA.Region})
+@inline function get_memory_region(region::HSA.Region, data::Ptr{HSA.Region}, kind)
     segment = Ref{HSA.RegionSegment}()
     HSA.region_get_info(region, HSA.REGION_INFO_SEGMENT, segment)
+    # Reject non-global regions
     if (segment[] != HSA.REGION_SEGMENT_GLOBAL)
         return HSA.STATUS_SUCCESS
     end
 
     flags = Ref{HSA.RegionGlobalFlag}()
+    host_accessible_region = Ref(false)
     HSA.region_get_info(region, HSA.REGION_INFO_GLOBAL_FLAGS, flags)
-    if (flags[] & HSA.REGION_GLOBAL_FLAG_KERNARG > 0)
+    HSA.region_get_info(region,
+        reinterpret(HSA.region_info_t, HSA.AMD_REGION_INFO_HOST_ACCESSIBLE),
```

Comment:
Btw, we now map such values to `HSA.RegionInfo` (see https://github.com/JuliaGPU/AMDGPU.jl/blob/ada175c31be4cad884c30b118cc8a29f0472b24b/src/hsa/HSA.jl#L61-L64)